pkg/meta/tkv.go

```diff
@@ -3756,3 +3758,146 @@ func (m *kvMeta) loadDumpedACLs(ctx Context) error {
 		return nil
 	})
 }
+
+type kvEntryScanner struct {
+	cache       []*Entry
+	cnt         int
+	m           *kvMeta
+	defaultCap  int
+	parentInode Ino
+	plus        bool
+	bgChan      chan []*Entry
+}
+
+func (m *kvMeta) NewMetaEntryScanner(inode Ino, plus bool) EntryScanner {
+	c := &kvEntryScanner{
+		cache:       make([]*Entry, 0),
+		cnt:         0,
+		m:           m,
+		parentInode: inode,
+		defaultCap:  DefaultCap,
+		plus:        plus,
+		bgChan:      nil,
+	}
+	return c
+}
+
+func (sc *kvEntryScanner) Valid() bool {
+	return true
+}
+
+func (sc *kvEntryScanner) Close() {
+	if sc.bgChan != nil {
+		_ = <-sc.bgChan
+		sc.cache = make([]*Entry, 0)
+		sc.cnt = 0
+		sc.bgChan = nil
+	}
+}
+
+func (sc *kvEntryScanner) scanEntries(startKey, endKey []byte, limit int) ([]*Entry, error) {
```

Comment:
change and reuse scanValues()?

Reply:
current, scanValues only take starKey, and use `NextKey(startKey)` as endKey, if a startKey is `fooD48` the endKey will be `fooD49`, but we want by endKey be `fooE` to retrieve any keys in `fooD48` to `fooE`.

It need to change `func (m *kvMeta) scanValues(prefix []byte, limit int, filter func(k, v []byte) bool) (map[string][]byte, error) ` to `func (m *kvMeta) scanValues(startKey, endKey []byte, limit int, filter func(k, v []byte) bool) (map[string][]byte, error) `

or Just add a new func  `func (m *kvMeta) scanValues2(startKey, endKey []byte, limit int, filter func(k, v []byte) bool) (map[string][]byte, error) `

Reply:
maybe like this
```
func (m *kvMeta) scanValues(prefix []byte, limit int, filter func(k, v []byte) bool) (map[string][]byte, error) {
	return m.doScanValues(prefix, nextKey(prefix), limit, filter)
}

func (m *kvMeta) doScanValues(startKey, endKey []byte, limit int, filter func(k, v []byte) bool) (map[string][]byte, error) {
	if limit == 0 {
		return nil, nil
	}
	values := make(map[string][]byte)
	err := m.client.txn(func(tx *kvTxn) error {
		var c int
		tx.scan(startKey, endKey, false, func(k, v []byte) bool {
			if filter == nil || filter(k, v) {
				values[string(k)] = v
				c++
			}
			return limit < 0 || c < limit
		})
		return nil
	}, 0)
	return values, err
}
```